### DISCUSSION TITLE
Any reason a service might stop processing events if it's state hasn't changed?

### DISCUSSION THREAD

**Discussion by Asderex** [statelyai/xstate#1721] [Q&A]
Hi,  I've found I have a bug (in my code) that is proving quite tricky to pinpoint and was hoping someone might have some suggestions as to what I might look at to track it down.  I am finding that one of machines stops responding to valid events being sent to it. 

I have three possible transitions defined on a single state (that it never leaves).  When I send events to the service from the console it behaves as expected most of the time.  However, every now and then the service appears to stop processing any events and doesn't recover without a refresh.  The `service.send` call to the service is returning the current state to my console but the `event` property appears to only show the last successful event - it feels like the service is paused somehow.

I wish I could produce a minimal working example but it is actually hard to recreate this issue in my own (now quite complex) code. Now, it's not a huge problem for me as I will likely be rebuilding the logic here anyway but as a learning exercise I was curious if there were any properties on a running service that will effect how an event is processed? Is there any internal state or flags within a service that prevent it from processing events being sent in that I could check?

I'm running v4.13.0

**Comment by davidkpiano** [comment]
✏️ Please create a [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed) reproduction of this. Thanks!

**Reply by Asderex** [reply]
Thanks David - I'll see if I can copy over the essense of the machine causing the problems but it is **super** intermittent (seemingly following the same steps does not necesarily raise the same issue).  The service in question is actually my "masterApplication" service which overseas all application state (within spawned actors) so getting a working example that triggers the effect I'm seeing in my code might be tough / impossible.   I understand that makes it impossible for you to provide any meaningful comment but I thought this might jog a "check if the ... property on the service is showing as ...". No worries if nothing springs to mind .  I'm sure once I flesh out a more thoughtful approach to saving and loading my services I can get around this.  I was actually as (if not more) interested in how I might solve the problem than solving the problem itself.
